pkg/cmd/server/server.go

```diff
@@ -440,6 +443,50 @@ func (s *server) run() error {
 	return nil
 }
 
+// setupBeforeControllerRun do any setup that needs to happen before the controllers are started.
+func (s *server) setupBeforeControllerRun() error {
+	client, err := ctrlclient.New(s.mgr.GetConfig(), ctrlclient.Options{Scheme: s.mgr.GetScheme()})
+	// the function is called before starting the controller manager, the embedded client isn't ready to use, so create a new one here
+	if err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to create client")
+		return err
+	}
+
+	markInProgressCRsFailed(s.ctx, client, s.namespace, s.logger)
+
+	if err := setDefaultBackupLocation(s.ctx, client, s.namespace, s.config.defaultBackupLocation, s.logger); err != nil {
+		return err
+	}
+	return nil
+}
+
+// setDefaultBackupLocation set the BSL that matches the "velero server --default-backup-storage-location"
+func setDefaultBackupLocation(ctx context.Context, client ctrlclient.Client, namespace, defaultBackupLocation string, logger logrus.FieldLogger) error {
+	if defaultBackupLocation == "" {
+		logger.Warnf("No default backup storage location specified. Velero will not automatically select a backup storage location for new backups.")
+		return nil
+	}
+
+	backupLocation := &velerov1api.BackupStorageLocation{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: defaultBackupLocation}, backupLocation); err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.WithField("backupStorageLocation", defaultBackupLocation).Info("Velero server specified a default backup storage location that does not exist.")
```

Comment:
Do you want to return directly when getting the not found error?

Reply:
I've removed `IsNotFound` condition. 

If users configured one non-exist BSL name as a default BSL we return one error more reasonable